Add setting for download directory

AmandaCameron opened the pull request:
This is mostly untested, but I think it should work. At the very least it's working in the case of the setting being defined.

AmandaCameron committed:
Add setting for download directory

flip-rossi commented:
Yup seems good!
I just think every setting should also be configurable in the menus without directly editing the config file. I can do this on Tuesday no problem, unless you'd like to give it a shot before that 😛

Would also be nice if changing this setting would move the previous directory to the new one

AmandaCameron committed:
settings: Add setting for download_dir

AmandaCameron commented:
Added a new field to the `Readeck server and credentials` dialog, and refactored the code so if `download_dir` is not set, or is an empty string, it'll default to the old one.

I'm not sure how I'd go about making it move the stuff when it's changed, though.

AmandaCameron committed:
settings: Switch to a top-level menu entry that pops the dialog used by opds, others

AmandaCameron commented:
Bah, that crashes.

AmandaCameron committed:
Fix references

AmandaCameron commented:
There, now there's a menu entry in the plugin's settings menu that'll pop the same UI used by OPDS, still doesn't do anything about moving the files, though
